### PR TITLE
Task 010

### PR DESCRIPTION
**Pull request by shiory602**
I create four circles.
Also, I tried to use display: flex, but it didn't work.
Could you tell me how to use the class row of layer 0 styles?

### CODE REVIEW

File: -sass/styles/s-icircle.scss [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    background-color: $primary-color;
+}
+
+.s-icircle:hover {
+    background-color: $primary-color-shade;
+}
+
+.s-icircle__icon {
+    text-align: center;
+    padding-top: 60px;
```
Comment (pmvald) [inline -sass/styles/s-icircle.scss:14]: probably margin is a better idea

File: tasks/task-010/index.html [hunk 2]
```diff
@@ -0,0 +1,69 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+    <link rel="stylesheet" href="../../font/bootstrap-icons.css">
+    <link rel="stylesheet" href="../../css/bundle.css">
```
Comment (pmvald) [inline tasks/task-010/index.html:9]: good

**Review by pmvald — CHANGES_REQUESTED** [review]
@shiory602 Good, almost there. A few small changes, plus: delete all the css files you made in the -sass folder.

File: -sass/styles/s-icircle.scss (hunk 1 above)
Comment (shiory602) [reply -sass/styles/s-icircle.scss]: May I ask you why better to use margen than padding?

Comment (pmvald) [reply -sass/styles/s-icircle.scss]: Padding is to *separate* a *wrapper* from its *content*. To add vertical space *between* blocks margin is the standard choice, or you can just add a vertical space using a spacer block, like this one which we included in our project:

```html
<div class="spacer-2"></div>
```

The above block adds a vertical space of size 2-rem (currently we have spacers from 1 to 4 rems). 

padding is usually used to separate a wrapper from its contents:

```html
<!-- wrapper block, with possibly a border -->
<div style="padding:4px; border: 1px solid grey"> 
  ... lot of content
</div>
```

The padding for wrapper is usually a full padding (=> includes all four directions) and mostly has the same value for all directions (like padding: 4px). The wrapper also usually has a border around it, so it basically adds a border with a nice padding around your content and allow you to group and organize your contents.

This was a good lecture! I'm going to add it to the design guide in fact ....

**Comment by pmvald** [comment]
@shiory602 I pushed some changes. Take a look at my code, and notice the changes, in particular:

* The file name had to be `_circles.scss` (learn: fragment scss files always start with _)
* you forgot to close some `<section>` tags
* I removed the top margin. In fact articles blocks should not have any margin. These spaces have to be added by the wrapping element.
* shadow/transition effects were added.
* `cursor: pointer` style to make the mouse become pointer when hover

Now, learn more about css transition by reading [this guide](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Transitions/Using_CSS_transitions). Now:
1. What transition I used in this design?
2. what is `ease-in-out`> what are other alternatives?
3. Can you also make the circle slightly bigger upon mouse hover? (if you know then do it!)
4. You may find some programmers use other css styles such as `-moz-transition`, `-o-transition` beside `transition`. What are these styles? Are they needed as well? (wanna see [this](http://www.web-plus-plus.com/Articles/css-transition-moz-and-webkit-vs-css3)).

**Comment by shiory602** [comment]
1. What transition I used in this design?
`transition: font-size 0.2s ease-in-out`
- transition-property to control the property "font-size".
- transition-duration to specify the changing duration.
- transition-timing-function to control the changing speed. 

2. what is ease-in-out> what are other alternatives?
- ease -> the default value. start and end slowly
- linear -> even speed
- ease-in -> start slowly and end quickly
- ease-out -> start quickly and end slowly
- ease-in-out -> start and end very slowly
[other alternatives](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Transitions/Using_CSS_transitions#defining_transitions)

3. Can you also make the circle slightly bigger upon mouse hover? (if you know then do it!)
I use `transition: transform 0.2s ease-in-out;` to `s-icircle`
Also, `transform: scale(1.5);` to `&:hover`

4. You may find some programmers use other css styles such as -moz-transition, -o-transition beside transition. What are these styles? Are they needed as well? (wanna see this).
-webkit- … Google Chrome, Sfari
-moz- … Firefox
-ms- … Internet Explorer
-o- … Opera
Those are "Bender Prefix". It is used to implement your own extensions that are not supported by the browser, or to implement a draft specification ahead of time.
You need to put it if the property is "Working Draft", but not need to put it if it is "Candidate Recommendation".
You can check if you need Bender Prefix [here](https://caniuse.com/)
And this time don't need to use it.

**Comment by pmvald** [comment]
@shiory602 Ok, now next step is to build this into a page. We want to follow the layout in [this page](https://velocity-app.newton.ca/public/#/clientJourney/slides/home?shortCode=1y2aiuoxz4mio&lang=en-ca). 

For the background picture you may use pic2 that I have attached (see if it looks nice, otherwise we can change it). Note that this picture is in .jpg format but I can get the adobe format as well if needed. But for now and for just a demo jpg should be good. 

For the main container (the window that says "welcome"), it is similar to bootstrap "container" but narrower. Check if bootstrap has some class like `container-md` (a container with md width which is approximately 992px). If not maybe we need to add these to our layouts (container-sm, container-md, etc.) Eitehr way, for the main container maybe we just use a container class with top margin (mt-4) and shadow (s-shadow class, but I thing this clas has to be defined in styles/_utilities using the variable $shadow).

Then inside the container follow the layout of pic1.jpg and make these changes/replacements:
1. The title real-estate => `<h2> ALL Vancouver properties here! </h2>`
2. the grey text under the header, leave it as it is. Use` <p class="s-thin s-mute">` for this text 
3. remove the icon part
4. The section with four circles is going to use the icircles articls that you designed. It is good to change the background to the gradient though (from $primary-color to $primary-shade-color ?). But remove the "Read more" text and replace it with "What type of property you are looking for?". The icon for each propety can be taken from the set of icons that you see in the picture. Icon titles are the same that we used in the previous demo (Condo, Townhouse, House, Land). Also keep the small description texts below each icon. FOr now just keep the Lorem text and we will replace it by a definition of these properties later. Note that it is possible to get the Adobe version of pic1.jpg if you think it is easier to work with, particularly to extract the icons you may need to use the adobe version. That would be nice if we can extract the icons as "svg" (not jpg). If you know how to do it I can gte the adobe version of this picture which hopefully allow us to extract the icons as svg, though I'm not sure



pic1
pic2

**Comment by shiory602** [comment]
I have a question.
Does the background image cover all body? or Can I use img tag to put the image?
Also, I tried to use "$primary-shade-color", but I got an error.

**Comment by pmvald** [comment]
@shiory602 
> I have a question.
> Does the background image cover all body? or Can I use img tag to put the image? 
The design deatils are up to you. In this case I think the background is better to cover the page by repeatition (like wallpaper). 
> Also, I tried to use "$primary-shade-color", but I got an error.
You need to follow the error message and see what the problem is. If the proble is this variable does not exist then check the `/styles/_variables.scss, maybe its name is different, like `$primary-color-shade` instead? I remeber there is some "shade" color variable in the _variables ...

**Comment by shiory602** [comment]
Thank you for answering! Now I fix the layout.

**Review by pmvald — APPROVED** [review]

**Comment by pmvald** [comment]
@shiory602 I did some refactoring and moved some styles to other files, in particular I made the class "ss-prop-selection-section" for the circles part (note that it is a layer 2 class, why?) Please take a moment and review the changes I made and let me know if you have any question.

**Comment by shiory602** [comment]
Sorry for my late reply. I had some problems with my computer. You create 2 classes for one layer because it is better for responsive design?
I have a question about the class name. So what does "ss" mean? I'm confused about how to set a class name.

**Comment by pmvald** [comment]
@shiory602 I explained the 3-layer css deign in the general guide. You may need to review it again. Layer 0 are those classses that are related to positioning and layout, mostly coming from boostrap like row, col-3, container, d-flex, etc. We use these layer 0 classes in all the projects WIHTOUT any change. Layer 1 and 2 classes are about "styling" (font-size, colors, font-family) and they are opinionated, meaning we need to change them for each project. But Layer 1 class are for those components that are very common in projects (like `s-icircle`, `s-panel`) and we need to use them in almost all the projects, albeit with some modification. Layer 2 are those components that are "very specific" to this current project (like `ss-prop-selection-section`) and we don't use them in other projects. We try to avoid this type of components as much as possible and use layer 1 components instead. But sometimes we need to define project specific styles. As for naming, layer 1 classes start with `s-`, layer 2 with `ss-` and layer 0 names can be any name (no restriction as we consider them part of the browser functionality)

**Comment by shiory602** [comment]
Thank you for the explanation! I'll also check the general guide again!

**Pull request merged**